packages/framework/fluid-static/src/rootDataObject.ts

```diff
@@ -165,10 +170,36 @@ const rootDataStoreId = "rootDOId";
  */
 export function createDOProviderContainerRuntimeFactory(props: {
 	schema: ContainerSchema;
+	compatMode: CompatMode;
 }): IRuntimeFactory {
-	return new DOProviderContainerRuntimeFactory(props.schema);
+	return new DOProviderContainerRuntimeFactory(props.schema, props.compatMode);
 }
 
+const compatModeRuntimeOptions: Record<CompatMode, IContainerRuntimeOptions> = {
```

Comment:
Do you think the number of features we need to disable for 1.X/enable for 2.X will continue to grow as 2.X development continues? If so, I'm wondering if it makes sense to give this object its own file if we will be continuously adding to it

Reply:
I think that would represent dangerous changes, since if some featureFoo is added in say 2.3 and is incompatible with 1.4, then it's probably also incompatible with 2.0 and it would be a semver break to turn it on in a minor release.

If there are any items that get added to this config I think they'd most likely have to be features already in 2.0 but off-by-default, getting turned on in 2.3 (such that 2.0 won't blow up when it sees them).

That said, seems reasonable to split it to a separate file - I've done that in latest.